src/Components/Map/Map.jsx

```diff
@@ -26,34 +28,56 @@ export default function Map() {
       zoom: zoom
     })
 
+    map.current.on('load', () => {
+      map.current.addSource('points', featureCollection)
+      map.current.addLayer(symbolLayer)
+    })
+
+    map.current.on('click', (event) => {
+      setCoordinates(event.lngLat)
+    })
   })
 
-  // stores new lat/long/zoom values when user interacts with a map
   useEffect(() => {
     if (!map.current) return;
 
-    map.current.on('move', () => { // on() method attaches event handlers for the selected elements and child elements
-      setLng(map.current.getCenter().lng.toFixed(4)) // getCenter() returns map geographical centerpoint {lng: 0, lat: 0}
-      setLat(map.current.getCenter().lat.toFixed(4)) // toFixed() defines number of digits following a decimal point, here it's 4 digits (0.0000)
-      setZoom(map.current.getZoom().toFixed(2)) // getZoom() returns the map's current zoom level as a number
+    map.current.on('move', () => {
+      setLng(map.current.getCenter().lng.toFixed(4))
+      setLat(map.current.getCenter().lat.toFixed(4))
+      setZoom(map.current.getZoom().toFixed(2))
     })
   })
 
-  // creates a single event listener to pull lat/long coordinates from the map on click
-  const getCoordinates = () => {
-    map.current.once('click', (event) => {
-      setCoordinates(event.lngLat)
-    })
+  useEffect(() => {
+    if (marker) marker.remove()
+
+    if (coordinates) {
+      const newMarker = new mapboxgl.Marker()
+      newMarker.setLngLat([coordinates.lng, coordinates.lat]).addTo(map.current)
+      setMarker(newMarker)
+    }
+  }, [coordinates])
+
```

Comment:
Nice! This seems like really clean logic, setting the coordinates up on line 36 and then triggering this useEffect when coordinates change in order to set the marker.